### PR TITLE
[DEV-1738] Add ApiDataListPageTemplate

### PR DESCRIPTION
**Pull request by MarBert**
#### List of Changes

Add ApiDataListPageTemplate 

#### Motivation and Context

- Adds the template to create the API List page for each product
- Requires [[DEV-1724] [DEV-1701]](https://github.com/pagopa/developer-portal/pull/923)

#### How Has This Been Tested?

With storybook:
http://localhost:6006/?path=/story/templates-apilistpagetemplate--showcase

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[DEV-1724]: https://pagopa.atlassian.net/browse/DEV-1724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEV-1701]: https://pagopa.atlassian.net/browse/DEV-1701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/storybook-app/public/icons/code.svg
Comment (MarcoPonchia) [inline apps/storybook-app/public/icons/code.svg:1]: There is a reason for having this icon duplicated instead using `apps/nextjs-website/public/icons/code.svg`

Comment (marcobottaro) [reply apps/storybook-app/public/icons/code.svg:1]: This icon is inside `apps/storybook-app/public/` folder which is now configured to be the directory of Storybook's public assets

**Review by MikeAtUqido — APPROVED** [review]

File: apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        title={hero.title}
+        subtitle={hero.subtitle}
+        theme={theme}
+        smallHeight={hero.heigth || '272px'}
+        titleVariant='h4'
+        gridTextSx={{
+          justifyContent: 'center',
+        }}
+      />
+      <Box paddingBottom={'80px'}>
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx:66]:
```suggestion
      <Box paddingBottom={6}>
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by tommaso1 — APPROVED** [review]

File: apps/storybook-app/stories/templates/ApiListPageTemplate.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { mockText, mockTextBlock } from '../mock-content.helper';
+
+const meta: Meta<typeof ApiListPageTemplate> = {
+  title: 'Templates/ApiListPageTemplate',
+  component: ApiListPageTemplate,
+};
+
+export default meta;
+
+export const Showcase: StoryObj<typeof ApiListPageTemplate> = {
```
Comment (marcobottaro) [inline apps/storybook-app/stories/templates/ApiListPageTemplate.stories.tsx:13]:
```suggestion
export const Showcase: StoryObj<typeof ApiDataListPageTemplate> = {
```

File: apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import EContainer from '@/editorialComponents/EContainer/EContainer';
+import ProductBreadcrumbs from '@/components/atoms/ProductBreadcrumbs/ProductBreadcrumbs';
+import { productPageToBreadcrumbs } from '@/helpers/breadcrumbs.helpers';
+import CardsGrid from '@/components/molecules/CardsGrid/CardsGrid';
+import { Box } from '@mui/material';
+import { Theme } from '@/editorialComponents/types/components';
+import { BannerLinkProps } from '@/components/atoms/BannerLink/BannerLink';
+import BannerLinks from '@/components/molecules/BannerLinks/BannerLinks';
+
+export type ApiListPageTemplateProps = {
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx:13]:
```suggestion
export type ApiDataListPageTemplateProps = {
```

File: apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    ctaLabel?: string;
+    href?: string;
+    icon: string;
+    tags?: { readonly label: string }[];
+  }[];
+  readonly bannerLinks: BannerLinkProps[];
+  readonly theme?: Theme;
+};
+
+const ApiListPageTemplate = ({
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/ApiListPageTemplate/ApiListPageTemplate.tsx:36]:
```suggestion
const ApiDataListPageTemplate = ({
```

File: .changeset/rare-files-punch.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+"nextjs-website": minor
+"storybook-app": minor
+---
+
+Add ApiListPageTemplate to the project
```
Comment (MarBert) [inline .changeset/rare-files-punch.md:6]:
```suggestion
Add ApiDataListPageTemplate to the project
```

File: apps/nextjs-website/src/components/templates/ApiDataListPageTemplate/ApiDataListPageTemplate.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        gridTextSx={{
+          justifyContent: 'center',
+        }}
+      />
+      <Box paddingBottom={6}>
+        <CardsGrid
+          cardVariant='outlined'
+          cards={cards}
+          cardSvg={true}
+          cardSize={{ xs: 12, md: 16 / Math.max(cards.length, 3) }}
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/ApiDataListPageTemplate/ApiDataListPageTemplate.tsx:71]: I don't think that this formula is what we need:
image

Comment (marcobottaro) [inline apps/nextjs-website/src/components/templates/ApiDataListPageTemplate/ApiDataListPageTemplate.tsx:71]:
```suggestion
          cardSize={{ xs: 12, md: 4 }}
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**